test/e2e/getenvoy_run_test.go

```diff
@@ -85,18 +83,32 @@ func requireEnvoyReady(t *testing.T, envoyWorkingDir string, stderr io.Reader, c
 	require.NoError(t, err, `error reading admin address file %q after running [%v]`, adminAddressPath, c)
 
 	log.Printf(`waiting for Envoy client to connect after running [%v]`, c)
-	envoyClient, err := utilenvoy.NewClient(string(adminAddress))
+	envoyClient, err := newClient(string(adminAddress))
 	require.NoError(t, err, `error from envoy client %s after running [%v]`, adminAddress, c)
 	require.Eventually(t, func() bool {
-		ready, e := envoyClient.IsReady()
+		ready, e := envoyClient.isReady()
 		return e == nil && ready
 	}, 1*time.Minute, 100*time.Millisecond, `envoy client %s never ready after running [%v]`, adminAddress, c)
 
 	return envoyClient
 }
 
 func verifyDebugDump(t *testing.T, workingDir string, c interface{}) {
-	debugArchive := envoytest.RequireRestoreWorkingDir(t, workingDir, c)
+	// Run deletes the debug store directory after making a tar.gz with the same name.
+	// Restore it so assertions can read the contents later.
+	debugArchive := filepath.Join(workingDir + ".tar.gz")
+	defer os.Remove(debugArchive) //nolint
+
+	src, err := os.Open(debugArchive)
+	require.NoError(t, err, "error opening %s after stopping [%v]", debugArchive, c)
+	zSrc, err := tar.NewDecompressor(debugArchive, src)
+	require.NoError(t, err, "error getting decompressor for %s after stopping [%v]", debugArchive, c)
+	if c, ok := zSrc.(io.Closer); ok {
```

Comment:
this is little confusing since only gzipped files enter this block.. shall we add comment here?

Reply:
made both a ReadCloser instead